metadata-ingestion/src/datahub/ingestion/source/sql/athena.py

```diff
@@ -558,9 +555,20 @@ def get_partitions(
             level=StructuredLogLevel.WARN,
         ):
             # We create an artifical concatenated partition key to be able to query max partition easier
-            part_concat = " || '-' || ".join(
-                self._casted_partition_key(key) for key in partitions
-            )
+            # Use CONCAT function with all VARCHAR-casted values to avoid type mismatches
```

Comment:
Moving this to a separate function would make it easier to test all possible scenarios and corner cases while no require mocking.